auslib/db.py

```diff
@@ -1040,11 +1060,33 @@ def addRelease(self, name, product, blob, changed_by, transaction=None):
     def updateRelease(self, name, changed_by, old_data_version, product=None, read_only=None, blob=None, transaction=None):
         if product or blob:
             self._proceedIfNotReadOnly(name, transaction=transaction)
+        where = [self.name == name]
         what = {}
-        if read_only is not None:
-            what['read_only'] = read_only
+
+        current_product = self.select(where=where, columns=[self.product], transaction=transaction)[0]["product"]
```

Comment:
It's an interesting point. I could see a potential use case for granting *only* access to mark releases as read only (regardless of product). However, I think this could be done by granting a permission of "release_read_only" with actions ["set"], and not setting product - because abscence of products grants access to all.